### ISSUE TITLE
[Bug]: interrupt a process by using OneSignal.User.addTags for await [only android]

### ISSUE THREAD

**Issue by Cojiroooo** [OneSignal/OneSignal-Flutter-SDK#789]
### What happened?

The latest version SDK causes interruption of a synchronous process when I use OneSignal.User.addTags or addTagWithKey for await.

### Steps to reproduce?

```Markdown
1. Install SDK v5.0.3
2. Launch the android app
3. Init OneSignal
4. Execute await OneSignal.User.addTags
```

### What did you expect to happen?

I expected to continue a process in synchronous after using OneSignal.User.addTags or addTagWithKey for await.

### OneSignal Flutter SDK version

5.0.3

### Which platform(s) are affected?

- [ ] iOS
- [X] Android

### Relevant log output

```Shell
No response
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by psatler** [comment]
@Cojiroooo is the issue you're facing related to this one https://github.com/OneSignal/OneSignal-Flutter-SDK/issues/776#issuecomment-1793513599 ?

**Comment by Cojiroooo** [comment]
@psatler Sorry for replying so late. 
It seems different from my issue. 
but I will check my app at the new version because that issue is fixed.

**Comment by nan-li** [comment]
Hi @Cojiroooo, thanks again for reporting this issue and your patience.

We have released the fix in [Release 5.1.0](https://github.com/OneSignal/OneSignal-Flutter-SDK/releases/tag/5.1.0).

Please update and reach out if you are still seeing issues. Thanks!
